src/hdmf/spec/spec.py

```diff
@@ -145,6 +145,17 @@ def build_const_args(cls, spec_dict):
     def __hash__(self):
         return id(self)
 
+    def path_str(self):
+        """
+        Return string showing full path to this spec, with spec names separated by '/', regardless of the spec type.
+        """
+        spec_path = self.name
+        spec = self.parent
+        while spec is not None:
+            spec_path = spec.name + '/' + spec_path
```

Comment:
I couple things...

- Not all specs have `name` set, so in the case of `data_type` specs (e.g. DynamicTable), this will be ambiguous. You might consider having the BaseStorageSpec constructor pass in `data_type_def` as `name`.
- Since parent can't be reassigned, shouldn't this value be constant i.e. it shouldn't need to be computed every time the function gets called?
- I think this should be a property. Is there a reason you didn't go that route?
- I prefer `path` to `path_str`. What do you think?

Reply:
> * Not all specs have `name` set, so in the case of `data_type` specs (e.g. DynamicTable), this will be ambiguous. You might consider having the BaseStorageSpec constructor pass in `data_type_def` as `name`.

Gotcha. It turns out there is already in `Validator.get_spec_loc` that does what `Spec.path_str` was trying to do, but takes into account cases with a `data_type_def` or `data_type_inc` and no name. So I moved that code into `Spec` and refactored `Validator.get_spec_loc(cls, spec)` to call it.  

> * Since parent can't be reassigned, shouldn't this value be constant i.e. it shouldn't need to be computed every time the function gets called?

Parent can be assigned after the spec is created. It's an odd case, but you can create a `GroupSpec` `A` with a path `'A'`. Spec `A` could be added to `GroupSpec` `B` and would then have path `'B/A'`. Spec `B` could be added to `GroupSpec` `C` and would then have path `'C/B/A'`, and so on.

> * I think this should be a property. Is there a reason you didn't go that route?

No reason. It does make sense as a property (getter only). I changed it. However, if it is not a constant, should it still be a property?

> * I prefer `path` to `path_str`. What do you think?

Agreed. I like `path` better. I changed it.